metagpt/rag/engines/simple.py

```diff
@@ -301,3 +310,23 @@ def _resolve_embed_model(embed_model: BaseEmbedding = None, configs: list[Any] =
     @staticmethod
     def _default_transformations():
         return [SentenceSplitter()]
+
+    @staticmethod
+    def _get_file_extractor() -> dict[str:BaseReader]:
+        """
+        Get the file extractor.
+        Currently, only PDF use OmniParse. Other document types use the built-in reader from llama_index.
+
+        Returns:
+            dict[file_type: BaseReader]
+        """
+        file_extractor: dict[str:BaseReader] = {}
+        if config.omniparse.base_url:
+            pdf_parser = OmniParse(
+                api_key=config.omniparse.api_key,
+                base_url=config.omniparse.base_url,
+                parse_options=OmniParseOptions(parse_type=OmniParseType.PDF, result_type=ParseResultType.MD),
```

Comment:
what if `input_files` are txts or docs in `SimpleDirectoryReader`.

Reply:
No impact, use `file_extractor` parameter of `SimpleDirectoryReader` to specify the parser of the corresponding file. Currently, only PDF will use `OmniParse`. Other files are still parsed by llamaindex built-in
```python
@staticmethod
def _get_file_extractor() -> dict[str:BaseReader]:
    """
    Get the file extractor.
    Currently, only PDF use OmniParse. Other document types use the built-in reader from llama_index.

    Returns:
        dict[file_type: BaseReader]
    """
    file_extractor: dict[str:BaseReader] = {}
    if config.omniparse.base_url:
        pdf_parser = OmniParse(
            api_key=config.omniparse.api_key,
            base_url=config.omniparse.base_url,
            parse_options=OmniParseOptions(parse_type=OmniParseType.PDF, result_type=ParseResultType.MD),
        )
        file_extractor[".pdf"] = pdf_parser

    return file_extractor
```

Reply:
```python
file_extractor = cls._get_file_extractor()
documents = SimpleDirectoryReader(
    input_dir=input_dir, input_files=input_files, file_extractor=file_extractor
).load_data()
```

Reply:
if you config with omniparse.base_url, you will use pdf_parser from OmniParse. But what if files are docs in input_files.

Reply:
`SimpleDirectoryReader` will determine the file type based on `input_files` and match the corresponding file parser. When using `OmniParse`, `file_extractor` specifies `file_extractor[".pdf"] = pdf_parser`, so only `.pdf` files will use `OmniParse`, and others will still use the default `llamaindex` parser. For details, see the `SimpleDirectoryReader.load_file` source code
```python
        file_suffix = input_file.suffix.lower()
        if file_suffix in default_file_reader_suffix or file_suffix in file_extractor:
            # use file readers
            if file_suffix not in file_extractor:
                # instantiate file reader if not already
                reader_cls = default_file_reader_cls[file_suffix]
                file_extractor[file_suffix] = reader_cls()
            reader = file_extractor[file_suffix]

            # load data -- catch all errors except for ImportError
            try:
                kwargs = {"extra_info": metadata}
                if fs and not is_default_fs(fs):
                    kwargs["fs"] = fs
                docs = reader.load_data(input_file, **kwargs)
```